From 4121815fb9bc977ab0953558a2a69fa33550c53c Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 14 May 2021 02:26:13 -0700
Subject: [PATCH] MyCom.h: fix build with gcc-11

* fixes:
  ./../src/lzma-sdk/C/7zip/Compress/LZMA/LZMAEncoder.h: In member function 'virtual ULONG NCompress::NLZMA::CEncoder::Release()':
  ./../src/lzma-sdk/C/7zip/Compress/LZMA/../../../Common/MyCom.h:159:32: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
    159 | STDMETHOD_(ULONG, Release)() { if (--__m_RefCount != 0)  \
        |                                ^~

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 src/lzma-sdk/C/Common/MyCom.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/src/lzma-sdk/C/Common/MyCom.h
+++ b/src/lzma-sdk/C/Common/MyCom.h
@@ -156,8 +156,7 @@ class CMyUnknownImp
 
 #define MY_ADDREF_RELEASE \
 STDMETHOD_(ULONG, AddRef)() { return ++__m_RefCount; } \
-STDMETHOD_(ULONG, Release)() { if (--__m_RefCount != 0)  \
-  return __m_RefCount; delete this; return 0; }
+STDMETHOD_(ULONG, Release)() { if (--__m_RefCount != 0) return __m_RefCount; delete this; return 0; }
 
 #define MY_UNKNOWN_IMP_SPEC(i) \
   MY_QUERYINTERFACE_BEGIN \
